bliss/encoder/metrics.py

```diff
@@ -149,55 +149,61 @@ def _get_postfix(self):
 
 
 class DetectionPerformance(FilterMetric):
-    """Calculates detection and classification metrics between two catalogs.
-    Note that galaxy classification metrics are only computed when the values are available in
-    both catalogs.
-    """
-
     def __init__(
         self,
-        mag_band: int = 2,
-        mag_bin_cutoffs: list = None,
-        bin_unit_is_flux: bool = False,
+        bin_cutoffs: list = None,
+        ref_band: int = 2,
+        bin_type: str = "Flux",
         exclude_last_bin: bool = False,
         filter_list: List[CatFilter] = None,
     ):
         super().__init__(filter_list if filter_list else [NullFilter()])
 
-        self.mag_band = mag_band
-        self.mag_bin_cutoffs = mag_bin_cutoffs if mag_bin_cutoffs else []
-        self.bin_unit_is_flux = bin_unit_is_flux
+        self.bin_cutoffs = bin_cutoffs if bin_cutoffs else []
+        self.ref_band = ref_band
+        self.bin_type = bin_type
         self.exclude_last_bin = exclude_last_bin
 
+        assert self.bin_type in {"Flux", "Mag"}, "invalid bin type"
```

Comment:
I'd call this `self.flux_bin_type`. Possibly values could include `nmgy`, `njy`, and `mag`.

Reply:
i.e., `nmgy`, `njy`, and `mag` are all measures of flux. Of the three, `mag` is the most vague because there are multiple magnitude scaled